pkg/syncer/fullsync.go

```diff
@@ -1561,41 +1563,151 @@ func cleanupCnsMaps(k8sPVs map[string]string, vc string) {
 	}
 }
 
-// cleanupUnusedPVCsAndSnapshotsFromGuestCluster iterates over all PVCs and VolumeSnapshots and
-// filter ones with DeletionTimestamp set but CNS specific finalizer present. These finalizers are added to
-// PVCs and VolumeSnapshots from guest cluster during their creation.
-// If such objects found, remove the finalizer from those objects if corresponding guest cluster is not-running/deleted.
-// This code handles cases where namespace deletion causes guest cluster and its corresponding components
-// to be deleted but associated objects on supervisor remain stuck in Terminating state.
-func cleanupUnusedPVCsAndSnapshotsFromGuestCluster(ctx context.Context) {
+func RemoveCNSFinalizerFromPVCIfTKGClusterDeleted(ctx context.Context, k8sClient clientset.Interface,
```

Comment:
Do we need to wait for full sync cycle to get this finalizer cleared?

Reply:
We should not call check TKC cluster is deleted or present each time for individual PVC? 
Ideally when TKC cluster is deleted, we should catch event and remove finalizers from PVCs.

Reply:
updated code with namespace reconciler as per offline discussion.